backend/src/user/user.service.ts

```diff
@@ -115,6 +131,38 @@ export class UserService {
     return res;
   }
 
+  /**
+   * Updates the username/email/refresh token hash of an existing user in the database
+   * @param id id of user to be updated
+   * @param fields updateable fields for user object (username, email, refresh token hash)
+   * @returns [`Err`, `User`]
+   */
+  async updateResetPassword(id: number, temporaryPassword: string) {
+    const hash = await argon2.hash(temporaryPassword);
+    const res = await radash.try(this.prisma.user.update)({
+      where: { id },
+      data: { hash },
+      select: USER_FIELDS,
+    });
+    return res;
+  }
+
+  /**
+   * Updates the username/email/refresh token hash of an existing user in the database
+   * @param id id of user to be updated
+   * @param fields updateable fields for user object (username, email, refresh token hash)
+   * @returns [`Err`, `User`]
+   */
+  async updatePasswordWithHash(id: number, newPassword: string) {
+    const hash = await argon2.hash(newPassword)
+    const res = await radash.try(this.prisma.user.update)({
+      where: { id },
+      data: { hash },
+      select: USER_FIELDS,
+    });
+    return res;
+  }
+
```

Comment:
are these functions necessary? think by adding the `hash` attribute to the `UpdateableUserFields` type, it'll allow the existing `update` function to accept password hashes as an attribute to be updated.

hence, the new `UpdateableUserFields` type would be:
```
type UpdateableUserFields = Partial<
  Pick<User, "username" | "email" | "hashRt" | "hash">
>;
```

Reply:
Thanks @aidanaden, I didn't use the original `update` function because I was unfamiliar with the syntax. I realised that I repeat the same function twice too 🤦 .